src/main/java/teammates/logic/core/Emails.java

```diff
@@ -808,4 +842,36 @@ private boolean isYetToJoinCourse(StudentAttributes s) {
         
         return data;
     }
+    
+    public Sendgrid parseMimeMessageToSendgrid(MimeMessage message) throws MessagingException, JSONException, IOException {
+        Sendgrid email = new Sendgrid(Config.SENDGRID_USERNAME, Config.SENDGRID_PASSWORD);
+        
+        for (int i = 0; i < message.getRecipients(Message.RecipientType.TO).length; i++) {
+            email.addTo(message.getRecipients(Message.RecipientType.TO)[i].toString());
+        }
+        
+        String from = message.getFrom()[0].toString();
+        String html = message.getContent().toString();
+        
+        // From name and email in the format: Name <Email>
+        if (from.contains("<") && from.contains(">")) {
+            from = from.substring(from.indexOf("<") + 1, from.indexOf(">"));
+        }
```

Comment:
This branch can be extracted into a static method? I noticed the same code appearing again in a test class below.